packages/react-scripts/template/README.md

```diff
@@ -1825,10 +1825,35 @@ Create React App doesn't provide any built-in functionality to publish a compone
 
 ## Making a Progressive Web App
 
-By default, the production build is a fully functional, offline-first
-[Progressive Web App](https://developers.google.com/web/progressive-web-apps/).
+The production build has all the tools necessary to generate a first-class
+[Progressive Web App](https://developers.google.com/web/progressive-web-apps/),
+but **the offline/cache-first behavior is opt-in only**. By default,
+the build process will generate a service worker file, but it will not be
+registered, so it will not take control of your production web app.
 
-Progressive Web Apps are faster and more reliable than traditional web pages, and provide an engaging mobile experience:
+In order to opt-in to the offline-first behavior, developers should look for the
+following in their [`src/index.js`](src/index.js) file:
+
+```js
+// If you want your app to work offline and load faster, you can change
+// unregister() to register() below. Note this comes with some pitfalls.
+// Learn more about service workers: http://bit.ly/CRA-PWA
+serviceWorker.unregister();
+```
+
+As the comment states, switching `serviceWorker.unregister()` to
+`serviceWorker.register()` will opt you in to using the service worker.
+
+>Note: In the current version of `create-react-app`, opting-in is required for
+all developers who want offline-first behavior, even if you've previously
+deployed a version of your web app which used a service worker. If you redeploy
```

Comment:
I don't think this is the right place to say redeploy because the opt-in only affect the people who generate a new project with CRA. The people that are upgrading to v2 from existing v1 generated project would still have the old registration code. (if they don't also update the file manually, it is)

Reply:
Ah, so the local `src/index.js` is left alone when someone updates `create-react-app` from v1 to v2? I've got to be honest that I'm not familiar with the upgrade process 😄 

If that's the case, I'll update the text. Thanks!

Reply:
Yes, all files in `src` are left alone when upgrading. "Upgrading" is really just updating a dependency in `package.json`. The template code in `src` is only used when creating a new project.

Reply:
usually, there would be a guide in the release like https://github.com/facebook/create-react-app/releases/tag/v1.0.8 to migrate the files in src, which I think is the more appropriate place for it.